Gems/ROS2/Code/Source/Lidar/ROS2LidarSensorComponent.cpp

```diff
@@ -100,6 +102,11 @@ namespace ROS2
             const TopicConfiguration& publisherConfig = m_sensorConfiguration.m_publishersConfigurations[PointCloudType];
             AZStd::string fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), publisherConfig.m_topic);
             m_pointCloudPublisher = ros2Node->create_publisher<sensor_msgs::msg::PointCloud2>(fullTopic.data(), publisherConfig.GetQoS());
+            if (m_lidarCore.m_lidarConfiguration.m_lidarSystemFeatures & LidarSystemFeatures::Segmentation) {
+                auto segmentationClassesPublisher = ros2Node->create_publisher<vision_msgs::msg::LabelInfo>(
```

Comment:
Are you sure it should not be:
```suggestion
                m_segmentationClassesPublisher = ros2Node->create_publisher<vision_msgs::msg::LabelInfo>(
```
Did you verifiy it?

Reply:
Good catch